tests/test_enterprise/views/test_grant_data_sharing_permissions.py

```diff
@@ -594,30 +604,120 @@ def test_post_course_specific_consent(
+            assert dsc.granted is True
 
-        # we'll only create an enrollment record if (1) creation is not deferred, (2) consent has not been provided,
-        # (3) we provide a license_uuid, and (4) we provide a course _run_ id
-        if not defer_creation and not consent_provided and license_uuid and course_id.endswith('Demo_Course'):
-            assert LicensedEnterpriseCourseEnrollment.objects.filter(
-                license_uuid=license_uuid,
+            assert EnterpriseCourseEnrollment.objects.filter(
+                enterprise_customer_user=ecu,
+                course_id=course_id,
             ).exists() is True
 
-            mock_enrollment_api_client.return_value.enroll_user_in_course.assert_called_once_with(
-                self.user.username,
+            if license_uuid:
+                assert LicensedEnterpriseCourseEnrollment.objects.filter(
+                    license_uuid=license_uuid,
+                ).exists() is True
+                mock_enrollment_api_client.return_value.enroll_user_in_course.assert_called_once_with(
+                    self.user.username,
+                    course_id,
+                    course_mode
+                )
+            else:
+                assert not mock_enrollment_api_client.return_value.enroll_user_in_course.called
+
+    @mock.patch('enterprise.views.render', side_effect=fake_render)
+    @mock.patch('enterprise.views.EnrollmentApiClient')
+    @mock.patch('enterprise.models.EnterpriseCatalogApiClient')
+    @mock.patch('enterprise.api_client.discovery.CourseCatalogApiServiceClient')
+    @mock.patch('enterprise.views.reverse')
+    @ddt.data(
+        (True, True, '/successful_enrollment', 'course-v1:edX+DemoX+Demo_Course', str(uuid.uuid4())),
+        (True, False, '/failure_url', 'course-v1:edX+DemoX+Demo_Course', str(uuid.uuid4())),
+        (False, True, '/successful_enrollment', 'course-v1:edX+DemoX+Demo_Course', str(uuid.uuid4())),
+        (False, False, '/failure_url', 'course-v1:edX+DemoX+Demo_Course', str(uuid.uuid4())),
+        (True, True, '/successful_enrollment', 'edX+DemoX', str(uuid.uuid4())),
+        (True, False, '/failure_url', 'edX+DemoX', str(uuid.uuid4())),
+        (False, True, '/successful_enrollment', 'edX+DemoX', str(uuid.uuid4())),
+        (False, False, '/failure_url', 'edX+DemoX', str(uuid.uuid4())),
+        (True, True, '/successful_enrollment', 'course-v1:edX+DemoX+Demo_Course', ''),
+        (True, False, '/failure_url', 'course-v1:edX+DemoX+Demo_Course', ''),
+        (False, True, '/successful_enrollment', 'course-v1:edX+DemoX+Demo_Course', ''),
+        (False, False, '/failure_url', 'course-v1:edX+DemoX+Demo_Course', ''),
+        (True, True, '/successful_enrollment', 'edX+DemoX', ''),
+        (True, False, '/failure_url', 'edX+DemoX', ''),
+        (False, True, '/successful_enrollment', 'edX+DemoX', ''),
+        (False, False, '/failure_url', 'edX+DemoX', ''),
+    )
+    @ddt.unpack
```

Comment:
Not necessarily feedback for this PR, because I know you're following the pattern of existing tests, but I feel like this is _not_ a good use case for `ddt` - 
* it's really hard to read and understand which parameter lines up with which tuple element
* I'm generally not a fan of mixing input with expected output like this
* It complicates the logic of the test body.

I'd rather split up the primary test conditions (is creation deferred, was consent provided?) into smaller tests with some common helper functions to set up all the test data we need.  I feel like that'll help make any future modifications of code and associated tests much easier to deal with.

Reply:
This test can be simplified to a single test case along the lines of "consent was previously provided, assert `_enroll_learner_in_course` was not called, assert the redirect was to `success_url`". We don't need all of these test cases for what is essentially a "take no action" scenario.

Reply:
Alrighty I can reduce it down to just the one test case